### PR TITLE
fix: java_ops issue with quoted values

### PR DESCRIPTION
**Pull request by jBouyoud**
# What this PR does / why we need it

Fix issues with quoted values like : `-Dhudson.model.DirectoryBrowserSupport.CSP="default-src 'self';"`

# Special notes for your reviewer

# Checklist
- [x] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] CHANGELOG.md was updated

### CODE REVIEW

**Comment by torstenwalter** [comment]
Can you add/adjust the unit tests to cater for the change?

**Comment by jBouyoud** [comment]
I'll be glad to do that but all test suite is failing. 
I've the same issue on my Desktop even on main branch, I tried to rollback to main with the same error (https://github.com/jenkinsci/helm-charts/pull/141/checks?check_run_id=1419552818) 

Tests are failing with : 
```
                - asserts[0] `isKind` fail
                        Template:       jenkins/templates/jenkins-master-deployment.yaml
                        DocumentIndex:  0
                        Expected to be kind:
                                Deployment
                        Actual:
                                null
```
I'm not very familiar with theses tools, and I can't find any info to be able to debug that.
Can you help me ? Is there any pro-tip to understand why the document is `null` ?

**Comment by torstenwalter** [comment]
I would try to run helm template with the values of the test case. Could be that there is an issue when rendering it.

**Comment by jBouyoud** [comment]
I've already tried this. 

For both `main` and my branch : 
- `helm template . -s templates/jenkins-master-deployment.yaml ` ✅  (output differ on two branch , pretty normal)
- `helm template . -s templates/jenkins-master-deployment.yaml | kaf - --dry-run=client ` ✅ 
- `helm unittest -f tests/jenkins-master-deployment-test.yaml .  ` ❌  with the error mentioned in the previous comment.
- Comment all `it` except `default values` ; run `helm unittest -f tests/jenkins-master-deployment-test.yaml .  ` ❌   with the same error

Environment (my desktop or CI) seems to be not relevant since I'm on MacOS and CI is executed on *nix platform and have the same issue

**Comment by torstenwalter** [comment]
@jBouyoud Looks like an error with the newest version of helm unittest. If #142 works then you can rebase your changed on top of it.

**Comment by torstenwalter** [comment]
```
				-        value: -Dcasc.reload.token=$(POD_NAME)
				+        value: "-Dcasc.reload.token=$(POD_NAME) \n"
```

**Comment by jBouyoud** [comment]
Tests ✅

File: charts/jenkins/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,10 @@ The change log until v1.5.7 was auto-generated based on git commits. Those entri
 
 Removed /tmp volume. Making /tmp a volume causes permission issues with jmap/jstack on certain Kubernetes clusters
 
+## 2.17.2
+
+Fix: `master.javaOpts` issue with quoted values
+
```
Comment (torstenwalter) [inline charts/jenkins/CHANGELOG.md:19]: Should be updated to 2.18.1 to reflect the chart version and moved above 2.18.0

**Review by torstenwalter — CHANGES_REQUESTED** [review]

**Review by torstenwalter — APPROVED** [review]

**Comment by torstenwalter** [comment]
@jBouyoud Thank you for this improvement!

**Pull request merged**